sway-error/src/error.rs

```diff
@@ -2430,6 +2433,18 @@ impl ToDiagnostic for CompileError {
                         },
                 }
             },
+            ConfigurableMissingAbiDecodeInPlace { span } => Diagnostic {
+                reason: Some(Reason::new(code(1), "Configurables need a function named \"abi_decode_in_place\" to be in scope".to_string())),
+                issue: Issue::error(
+                    source_engine,
+                    span.clone(),
+                    String::new()
+                ),
+                hints: vec![],
+                help: vec![
+                    "Function \"abi_decode_in_place\" is defined in the standard library module \"core::codec\".".into()
```

Comment:
Hmm, having just this first sentence in the help without the second proposed one that suggests to verify if the function is actually there, this help actually becomes confusing because it states that the function _is defined_ in the "code::codec" and the error is there very likely because it is _not_ defined.

There is another one issue. The error is currently emitted on every configurable because the span points to an individual configurable. It should be just a single error pointing to either the whole `configurable` block or to the text configurable, because it is not specific to a particular configurable.

Together with the above help message it results in actually a confusing experience.

Coming to an optimally crafted diagnostic is never easy :smile: and usually takes some polishing. After looking once again at the overall formulation I propose the following changes.

We leave the old-style single line formulation in line 967 as it is.

Reason should answer why is this an issue. The fact that "Configurables need a function named \"abi_decode_in_place\" to be in scope" is not an issue that's how it should be :smile: 

So, to be consistent with the guidelines, proposal is to reformulate the `Reason` as:  Function \"abi_decode_in_place\" is not in scope

`Issue::error` should actually point at the `configurable` block or even better at the span of the "configurable" and further explain: Configurables need a function named \"abi_decode_in_place\" to be in scope. // With the fullstop this time :smile:

Note that we can still emit the error for every configurable. That's fine because the deduplication will remove duplicates and leave only a single error message. 

And finally in the help we have both sentences, to guide where to look:
"Function \"abi_decode_in_place\" is usually defined in the standard library module \"core::codec\".".into(),
"Verify that you are using a version of the \"core\" standard library that contains this function.".into(),

Note also the added word "usually" because we allow developers to define their own encoding/decoding mechanisms by implementing functions in scope.

Reply:
> There is another one issue. The error is currently emitted on every configurable because the span points to an individual configurable. It should be just a single error pointing to either the whole configurable block or to the text configurable, because it is not specific to a particular configurable.

Not sure I understood this point. What is not being used is the configurable, not the block. Why highlighting the whole block would be better?

![image](https://github.com/FuelLabs/sway/assets/83425/547cb198-c3bb-47a0-82e0-3057b37bce1a)

Reply:
The warning is perfectly fine, of course :smile:  I was talking about the _error_ which is not related to a particular configurable and will be rendered on every configurable. Currently for e.g.:
```Sway
configurable {
    MAX_SUPPLY: u64 = 3,
    OTHER: u64 = 3,
    MORE: u64 = 3,
}
```
rendered errors will be quite overwhelming and confusing:

![Configurables need a function named abi_decode_in_place to be in scope](https://github.com/FuelLabs/sway/assets/4142833/abc2b9db-cdba-4771-b84f-9a38807bd694)

The impression those message give is, that there is something problematic with an individual configurable, but they are all perfectly fine and there is nothing to change about them. It is actually a piece of "infrastructure" that is missing. That's why I think having a single error (duplicates automatically removed via deduplication) pointing to the block will better communicate the issue.

Reply:
Ah. In this case it makes sense.

Reply:
I will use the keyword span, as the whole block renders very strangely. 
![image](https://github.com/FuelLabs/sway/assets/83425/adcc81c2-f0f4-4d64-8d91-249a8eb8592f)

Reply:
Yes unfortunately this issue with code snippets is a know one, #5499, but still open. Using keyword is also fine.